packages/flutter/lib/src/cupertino/magnifier.dart

```diff
@@ -0,0 +1,308 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+import 'package:flutter/widgets.dart';
+
+/// A [CupertinoMagnifier], specifically for integrating with [SelectionOverlay].
+///
+/// Delegates styling to [CupertinoMagnifier] with its position depending on
+/// [magnifierOverlayInfoBearer].
+///
+/// Specifically, the [CupertinoTextMagnifier] follows the following rules.
+/// [CupertinoTextMagnifier]:
+/// - is positioned horizontally outside the screen width, with [horizontalScreenEdgePadding] padding.
+/// - is hidden if a gesture is detected [hideBelowThreshold] units below the line
+///   that the magnifier is on, shown otherwise.
+/// - follows the x coordinate of the gesture directly (with respect to rule 1).
+/// - has some vertical drag resistance; i.e. if a gesture is detected k units below the field,
+///   then has vertical offset [dragResistance] * k.
+class CupertinoTextMagnifier extends StatefulWidget {
+  /// Construct a [RawMagnifier] in the Cupertino style, positioning with respect to
+  /// [magnifierOverlayInfoBearer].
+  ///
+  /// The default constructor parameters and constants were eyeballed on
+  /// an iPhone XR iOS v15.5.
+  const CupertinoTextMagnifier({
+    super.key,
+    required this.controller,
+    required this.magnifierOverlayInfoBearer,
+    this.dragResistance = 10.0,
+    this.hideBelowThreshold = -48.0,
+    this.horizontalScreenEdgePadding = 10.0,
+    this.animationCurve = Curves.easeOut,
+  });
+
+  /// The curve used for the in / out animations.
+  final Curve animationCurve;
+
+  /// A drag resistance on the downward Y position of the lens.
+  final double dragResistance;
+
+  /// The difference in Y between the gesture position and the caret center
+  /// so that the magnifier hides itself.
+  final double hideBelowThreshold;
+
+  /// The padding on either edge of the screen that any part of the magnifier
+  /// cannot exist past.
+  ///
+  /// This includes the entire magnifier, not just the center.
```

Comment:
Nit: Is this more clear? "This includes any part of the magnifier, not just the center."

Reply:
Done and added more elaboration since it maybe was still not good enough